healthScore/tests.py

```diff
@@ -450,4 +451,143 @@ def test_registration_user_username_exists_error(self):
         self.assertEqual(response.status_code, 500)
 
 
-# Create your tests here.
+class HomepageViewTest(TestCase):
+    def test_homepage_view(self):
+        response = self.client.get(reverse("homepage"))  # test the view
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "homepage.html")
+
+
+class LoginViewTest(TestCase):
+    def setUp(self):
```

Comment:
Instead of a different class, could we use the already existing class since it had all the DB data setup?

Reply:
yes, your idea seems to follow the DRY principle in terms of adding data. For unit tests, I think a clearer way is to write unit tests for each API in views.py or class in models.py. In this way, we can identify and modify all unit tests for each api according to the test case class name. Meanwhile, since each unit test is only for one unit feature, it won't be bothered adding too much data.

here is a resource I found useful for how to write a unit test: https://realpython.com/testing-in-django-part-1-best-practices-and-examples/

Reply:
This is resolved!